### PR TITLE
[DOC] Some changes to functions.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: docs/contracts/functions.rst [hunk 1]
```diff
@@ -286,27 +286,40 @@ Below you can see an example of a Sink contract that uses function ``receive``.
...
 
 .. note::
     Even though the fallback function cannot have arguments, one can still use ``msg.data`` to retrieve
     any payload supplied with the call.
+    After having checked the first four bytes of ``msg.data``,
+    you can use ``abi.decode`` together with the array slice syntax to
+    decode ABI-encoded data:
+    ``(c, d) = abi.decode(msg.data[4:], (uint256, uint256));``
+    Note that this should only be used as a last resort and
+    proper functions sould be used instead.
```
Comment (leonardoalt) [inline docs/contracts/functions.rst]:
```suggestion
    proper functions should be used instead.
```

**Review by leonardoalt — COMMENTED** [review]
Looks good apart from typo

File: docs/contracts/functions.rst [hunk 2]
```diff
@@ -286,27 +286,40 @@ Below you can see an example of a Sink contract that uses function ``receive``.
...
     to distinguish Ether transfers from interface confusions.
 
 .. note::
     Even though the fallback function cannot have arguments, one can still use ``msg.data`` to retrieve
     any payload supplied with the call.
+    After having checked the first four bytes of ``msg.data``,
+    you can use ``abi.decode`` together with the array slice syntax to
+    decode ABI-encoded data:
+    ``(c, d) = abi.decode(msg.data[4:], (uint256, uint256));``
+    Note that this should only be used as a last resort and
```
Comment (ChrisChinchilla) [inline docs/contracts/functions.rst:320]: Ideally make this an actual Sphinx note, but can add that later as well.

**Review by ChrisChinchilla — COMMENTED** [review]
One small formatting suggestion

File: docs/contracts/functions.rst (hunk 2 above)
Comment (chriseth) [reply docs/contracts/functions.rst:320]: It already is inside a note...

**Pull request merged**